src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java

```diff
@@ -1,35 +1,95 @@
 package edu.kis.vh.nursery;
 
+class Node {  //in case of problems
+
+    public int getValue() {
+        return value;
+    }
+
+
+    private int value;
+    private Node prev, next;
+
+    public Node(int i) {
+        value = i;
+    }
+
+    public Node getPrev() {
+        return prev;
+    }
+
+    public void setPrev(Node prev) {
+        this.prev = prev;
+    }
+
+    public Node getNext() {
+        return next;
+    }
+
+    public void setNext(Node next) {
+        this.next = next;
+    }
+
+}
+
 public class DefaultCountingOutRhymer {
 
-    public int total = -1;
-    private int[] numbers = new int[12];
+    private static final int MAX_SIZE = 12;
+
+    private final int[] numbers = new int[getMaxSize()];
+
+    private int total = getDefIndexStack();
+
+    public static int getMaxSize() {
+        return MAX_SIZE;
+    }
+
+    public static int getDefIndexStack() {
+        return DEF_INDEX_STACK;
+    }
+
+    public static int getNotFound() {
+        return NOT_FOUND;
+    }
+
+    public int getTotal() {
+        return total;
+    }
+
+    private static final int DEF_INDEX_STACK = -1;
+
+    private static final int NOT_FOUND = -1;
 
     public void countIn(int in) {
         if (!isFull())
-            numbers[++total] = in;
+            getNumbers()[++total] = in;
     }
 
-    public boolean callCheck() {
+    protected boolean callCheck() {
 
-        return total == -1;
+        return total == getDefIndexStack();
     }
 
-    public boolean isFull() {
+    protected boolean isFull() {
 
-        return total == 11;
+        return total == getMaxSize() -1;
     }
 
     protected int peekaboo() {
         if (callCheck())
-            return -1;
-        return numbers[total];
+            return getNotFound();
+        return getNumbers()[total];
     }
 
-    public int countOut() {
+    protected int countOut() {
         if (callCheck())
             return -1;
-        return numbers[total--];
+        return getNumbers()[total--];
+    }
+
+    public int[] getNumbers() {
```

Comment:
Shouldn't be encapsulated (typical private field). This is an exaggeration. Using private fields internally usually do not require encapsulation.